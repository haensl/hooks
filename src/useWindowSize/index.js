import { useState, useEffect } from 'react';
import { useDebounce } from '../';

const useWindowSize = (debounceMs = 25) => {
  const [size, setSize] = useState(null);

  const onResize = useDebounce(() => {
    setSize({
      width: window.innerWidth,
      height: window.innerHeight
    });
  }, debounceMs);

  useEffect(() => {
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [onResize]);

  return size;
};

export default useWindowSize;
