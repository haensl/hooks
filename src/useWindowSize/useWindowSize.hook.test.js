import { act, renderHook } from '@testing-library/react';
import '@testing-library/jest-dom';
import useWindowSize from './';

describe('useWindowSize', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  test('renders without crashing', () => {
    const { result } = renderHook(() => useWindowSize());
    expect(result.error)
      .not
      .toBeDefined();
  });

  describe('when the window is resized', () => {
    it('propagates the window size', () => {
      const { result } = renderHook(() => useWindowSize());

      act(() => {
        window.innerHeight = 200;
        window.innerWidth = 150;
        window.dispatchEvent(new Event('resize'));
        jest.runOnlyPendingTimers();
      });

      expect(result.current.width)
        .toEqual(150);
      expect(result.current.height)
        .toEqual(200);
    });
  });
});
