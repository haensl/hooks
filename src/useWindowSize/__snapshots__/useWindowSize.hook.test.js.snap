// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useWindowSize renders as expected 1`] = `
<span>
  no window size
</span>
`;

exports[`useWindowSize when the window is resized renders as expected 1`] = `
<TestComponent>
  <span>
    150,200
  </span>
</TestComponent>
`;
